File: uts 2306146 AI take home test/UTS soal 1.py
def sistem_pakar():
    print("Jawab pertanyaan berikut dengan 'ya' atau 'tidak':")

    R1 = input("Apakah daun menguning? ").lower() == 'ya'
    R2 = input("Apakah terdapat bercak hitam? ").lower() == 'ya'
    R3 = input("Apakah daun berlubang? ").lower() == 'ya'
    R4 = input("Apakah tanaman layu? ").lower() == 'ya'

    if R1 and R4:
        print("Jenis hama terdeteksi: Wereng (A1)")
    elif R2 and R1:
        print("Jenis hama terdeteksi: Jamur Daun (A2)")
    elif R3 and R2:
        print("Jenis hama terdeteksi: Ulat Grayak (A3)")
    elif R1 and R3:
        print("Jenis hama terdeteksi: Aphid (A4)")
    else:
        print("Gejala tidak cocok dengan hama yang diketahui.")

sistem_pakar()

File: uts 2306146 AI take home test/UTS soal 1.pl
% Fakta: R1 - Daun menguning
% Fakta: R2 - Terdapat bercak hitam
% Fakta: R3 - Daun berlubang
% Fakta: R4 - Tanaman layu

% Rules (Aturan):
% Jika R1 dan R4, maka A1 (Hama Wereng)
hama(A1) :- gejala(R1), gejala(R4).

% Jika R2 dan R1, maka A2 (Jamur Daun)
hama(A2) :- gejala(R2), gejala(R1).

% Jika R3 dan R2, maka A3 (Ulat Grayak)
hama(A3) :- gejala(R3), gejala(R2).

% Jika R1 dan R3, maka A4 (Hama Aphid)
hama(A4) :- gejala(R1), gejala(R3).

% Menampilkan deskripsi hama
deskripsi(A1) :- write('Hama terdeteksi: Wereng').
deskripsi(A2) :- write('Hama terdeteksi: Jamur Daun').
deskripsi(A3) :- write('Hama terdeteksi: Ulat Grayak').
deskripsi(A4) :- write('Hama terdeteksi: Aphid').

% Program utama
start :-
    tanya_gejala,
    (   hama(Hama) -> deskripsi(Hama)
    ;   write('Tidak dapat mengidentifikasi hama berdasarkan gejala yang diberikan.')
    ).

% Tanya gejala ke pengguna
tanya_gejala :-
    tanya('Apakah daun menguning? (ya/tidak)', R1),
    tanya('Apakah terdapat bercak hitam? (ya/tidak)', R2),
    tanya('Apakah daun berlubang? (ya/tidak)', R3),
    tanya('Apakah tanaman layu? (ya/tidak)', R4).

% Menyimpan gejala yang dijawab 'ya'
tanya(Pertanyaan, Gejala) :-
    write(Pertanyaan), nl,
    read(Jawaban),
    (   Jawaban == ya -> assert(gejala(Gejala))
    ;   true).

